feat: added Dockerfile (#72)

tmm360 opened the pull request.

vojtechsimetka commented:
This looks good to me, WDYT @nugaon ?

nugaon commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+RUN npm run build
+
+FROM node:15.14-alpine as final
+RUN yarn global add serve
```
why do we use `yarn` here instead of `npm`?

nugaon replied:
This whole setup could happen by running only `npx serve` instead. I don't see justified this additional layer.

tmm360 replied:
I'm following the readme instructions. Will try to use `npx serve`

nugaon replied:
> The quickest way to get started is to just run npx serve in your project's directory.
https://www.npmjs.com/package/serve#usage

This is the first sentence in the readme instructions, after that just an other way is explained how to do it.

tmm360 replied:
Running with `npx serve` it says:
`npm WARN exec The following package was not found and will be installed: serve`

So it's to choose if install it with container, or with first execution. My opinion, best to include before. A container should work as is, minimizing requirements to external resources to run. More predictable.

nugaon replied:
yeah I see now, you have only the `build` folder. you are right, I just wanted to propose to use only one package manager in the same workflow, can you try with this please?
```suggestion
RUN npm i -g serve
```

vojtechsimetka replied:
Thanks @tmm360 and @nugaon , we just merged a PR which adds serve as devDependency so that users who want to use it directly outside of Docker have simpler life. I believe it also solves this discussion. Cheers :) https://github.com/ethersphere/bee-dashboard/pull/79

tmm360 replied:
@vojtechsimetka Unfortunately because `serve` is imported as a dev dependency, and I'm optimizing image exporting only build result, serve isn't available on the final image.
Instead "RUN npm i -g serve" works. At this point I'm going to take this last one.

nugaon commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+RUN npm ci
+RUN npm run build
+
+FROM node:15.14-alpine as final
```
let's be unified 
```suggestion
FROM node:15.14-alpine AS final
```

nugaon commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+RUN yarn global add serve
+WORKDIR /app
+COPY --from=build /src/build .
+EXPOSE 5000
```
I can't see anywhere it would run on different port than the default `3000`, or am I wrong?

tmm360 replied:
It's exposed on 5000

tmm360 commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+WORKDIR /app
+COPY --from=build /src/build .
+EXPOSE 5000
+ENTRYPOINT ["serve"]
```
It works also without to me, but can add, no problem

vojtechsimetka commented:
I am very sorry @tmm360 but we are making yet another changes how the app is served. See #88 . I presume this might affect the PR.

tmm360 commented:
> I am very sorry @tmm360 but we are making yet another changes how the app is served. See #88 . I presume this might affect the PR.

Probably yes, I will wait for #88 to be merged for update this. Thanks

tmm360 commented:
The built version inside `./build` directory doesn't report the `package.json` file, required by the `npm run serve` command, so for Dockerfile I still use the `serve` command installed globally.

Anyway the `serve` command exposes by default on port `5000`, so I've updated it to `8080` for standardize with documentation.

I've also updated readme instructions for a local build with Docker, until image will be published on docker hub.

nugaon commented on README.md:
````diff
@@ -45,6 +45,15 @@ npm run serve
 
+## Docker
+
+From inside project directory:
+
+```sh
+docker build . -t bee-dashboard
+docker run -p 8080:8080 bee-dashboard
+```
````
```suggestion
To build Docker image, execute the following from inside project directory:

```sh
docker build . -t bee-dashboard
docker run --rm -p 127.0.0.1:8080:8080 bee-dashboard
```
```

tmm360 replied:
Missing the code block closing marker ```

vojtechsimetka replied:
It's not, but github thinks it is the opening of next codeblock :/

nugaon reviewed (approved):
LGTM, Thanks for the contrib! : )

after merging this, we can create the corresponding CI PR to publish this image under `ethersphere` IMO
